### ISSUE TITLE
aws_dms_replication_task can't use template files for replication_task_settings

### ISSUE THREAD

**Issue by ebower12** [terraform-providers/terraform-provider-aws#7641]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

* provider.archive: version = "~> 1.1"
* provider.aws: version = "~> 1.59"
* provider.external: version = "~> 1.0"
* provider.null: version = "~> 2.0"
* provider.template: version = "~> 2.0"

### Affected Resource(s)

`aws_dms_replication_task`

### Terraform Configuration Files

```
data "template_file" "gateway-payments-mappings" {
  template = "${file("${path.module}/table-mappings/gateway-payments.json")}"
}

data "template_file" "gateway-payments-settings" {
  template = "${file("${path.module}/task-settings/gateway-payments.json")}"
}

resource "aws_dms_replication_task" "gateway-payments" {
  replication_task_id       = "${var.env}-gateway-payments"
  replication_instance_arn  = "${aws_dms_replication_instance.gateway-replication.replication_instance_arn}"
  migration_type            = "cdc"
  source_endpoint_arn       = "${aws_dms_endpoint.source-endpoint.endpoint_arn}"
  target_endpoint_arn       = "${data.external.target-endpoint-arn.result["result"]}"
  table_mappings            = "${data.template_file.gateway-payments-mappings.rendered}"
  replication_task_settings = "${data.template_file.gateway-payments-settings.rendered}"
}
```

task-settings/gateway-payments.json for example is

```
"{\"TargetMetadata\":{\"TargetSchema\":\"\",\"SupportLobs\":true,\"FullLobMode\":false,\"LobChunkSize\":0,\"LimitedSizeLobMode\":true,\"LobMaxSize\":16,\"InlineLobMaxSize\":0,\"LoadMaxFileSize\":0,\"ParallelLoadThreads\":0,\"ParallelLoadBufferSize\":0,\"BatchApplyEnabled\":false,\"TaskRecoveryTableEnabled\":false},\"FullLoadSettings\":{\"TargetTablePrepMode\":\"DO_NOTHING\",\"CreatePkAfterFullLoad\":false,\"StopTaskCachedChangesApplied\":false,\"StopTaskCachedChangesNotApplied\":false,\"MaxFullLoadSubTasks\":8,\"TransactionConsistencyTimeout\":600,\"CommitRate\":10000},\"Logging\":{\"EnableLogging\":true,\"LogComponents\":[{\"Id\":\"SOURCE_UNLOAD\",\"Severity\":\"LOGGER_SEVERITY_DEFAULT\"},{\"Id\":\"TARGET_LOAD\",\"Severity\":\"LOGGER_SEVERITY_DEFAULT\"},{\"Id\":\"SOURCE_CAPTURE\",\"Severity\":\"LOGGER_SEVERITY_DEFAULT\"},{\"Id\":\"TARGET_APPLY\",\"Severity\":\"LOGGER_SEVERITY_DEFAULT\"},{\"Id\":\"TASK_MANAGER\",\"Severity\":\"LOGGER_SEVERITY_DETAILED_DEBUG\"}],\"CloudWatchLogGroup\":\"dms-tasks-staging-gateway-replication\",\"CloudWatchLogStream\":\"dms-task-NK5C6EBCKAGYL7XTZRIDKJ5ULA\"},\"ControlTablesSettings\":{\"historyTimeslotInMinutes\":5,\"ControlSchema\":\"\",\"HistoryTimeslotInMinutes\":5,\"HistoryTableEnabled\":false,\"SuspendedTablesTableEnabled\":false,\"StatusTableEnabled\":false},\"StreamBufferSettings\":{\"StreamBufferCount\":3,\"StreamBufferSizeInMB\":8,\"CtrlStreamBufferSizeInMB\":5},\"ChangeProcessingDdlHandlingPolicy\":{\"HandleSourceTableDropped\":false,\"HandleSourceTableTruncated\":false,\"HandleSourceTableAltered\":false},\"ErrorBehavior\":{\"DataErrorPolicy\":\"LOG_ERROR\",\"DataTruncationErrorPolicy\":\"LOG_ERROR\",\"DataErrorEscalationPolicy\":\"SUSPEND_TABLE\",\"DataErrorEscalationCount\":0,\"TableErrorPolicy\":\"SUSPEND_TABLE\",\"TableErrorEscalationPolicy\":\"STOP_TASK\",\"TableErrorEscalationCount\":0,\"RecoverableErrorCount\":-1,\"RecoverableErrorInterval\":5,\"RecoverableErrorThrottling\":true,\"RecoverableErrorThrottlingMax\":1800,\"ApplyErrorDeletePolicy\":\"IGNORE_RECORD\",\"ApplyErrorInsertPolicy\":\"LOG_ERROR\",\"ApplyErrorUpdatePolicy\":\"LOG_ERROR\",\"ApplyErrorEscalationPolicy\":\"LOG_ERROR\",\"ApplyErrorEscalationCount\":0,\"ApplyErrorFailOnTruncationDdl\":false,\"FullLoadIgnoreConflicts\":true,\"FailOnTransactionConsistencyBreached\":false,\"FailOnNoTablesCaptured\":false},\"ChangeProcessingTuning\":{\"BatchApplyPreserveTransaction\":true,\"BatchApplyTimeoutMin\":1,\"BatchApplyTimeoutMax\":30,\"BatchApplyMemoryLimit\":500,\"BatchSplitSize\":0,\"MinTransactionSize\":1000,\"CommitTimeout\":1,\"MemoryLimitTotal\":1024,\"MemoryKeepTime\":60,\"StatementCacheSize\":50},\"PostProcessingRules\":null}"
```

### Debug Output

```
* module.dms.aws_dms_replication_task.gateway-payments: 1 error(s) occurred:

aws_dms_replication_task.gateway-payments: InvalidParameterValueException: Replication Task Settings document error: Invalid json
        status code: 400, request id: ddc077f1-3607-11e9-85a8-c1dacf7001e7
```

### Panic Output

None

### Expected Behavior

Terraform builds the replication task and uses the string from the JSON file as the value for the `replication_task_settings`

### Actual Behavior

Terraform doesn't recognize the string as valid JSON and crashes

I've tried pointing the `replication_task_settings` directly to the json file instead of using a `template_file`, I've tried doing a `jsonencode()` on the json file, I've tried setting the json string as an EOF in the template file data source. None of these were recognized as valid JSON.

### Steps to Reproduce

1.Create a `aws_dms_replication_task` and assign the `replication_task_settings` property as the rendered value of a `template_file` that points to a .json file
2. Terraform apply

### Important Factoids

None

### References

None

**Comment by landesjm** [comment]
@ebower12 - thanks for posting this issue which I used as a template. However I was able to get the DMS task generated. The differences are my version of provider.aws is new and my task-settings json file is not escaped, just a regular json file.   

Terraform Version: 
Terraform v0.11.11
+ provider.aws v2.1.0
+ provider.null v2.1.0
+ provider.random v2.0.0
+ provider.template v2.1.0
+ provider.vault v1.6.0

Hope this helps.
